scripts/mod_loader/modui/weapon_deck_selector.lua

```diff
@@ -0,0 +1,470 @@
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_DisableAll_Title")),
+			})
+			:addTo(buttonLayout)
+		function disableAllButton.onclicked()
+			for _, button in ipairs(buttons) do
+				button.checked = false
+			end
+			enableSaveLoad(true)
+			return true
+		end
+
+		-- add spacer before preset buttons
+		-- width is crafted to right align the preset buttons
+		Ui()
+			:widthpx(frametop.w													-- button area width
+				- buttonLayout.padl - buttonLayout.padr		-- padding on sides
+				- WEAPON_WIDTH * (PRESET_WIDTH + size * 3) -- all buttons, 2 are half sized, one fixed
+				- buttonLayout.gapHorizontal * 5)					-- gap between buttons
+			:heightpx(BUTTON_HEIGHT):addTo(buttonLayout)
+		-- preset dropdown
+		local presetDropdown = UiDropDown(presets)
+			:widthpx(WEAPON_WIDTH * PRESET_WIDTH):heightpx(BUTTON_HEIGHT)
+			:settooltip(GetText("ConfigureWeaponDeck_Preset_Tooltip"))
+			:decorate({
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_Preset_Title")),
+				DecoDropDownText(nil, nil, nil, 33),
+				DecoAlign(0, -2),
+				DecoDropDown(),
+			})
+			:addTo(buttonLayout)
+		function presetDropdown:destroyDropDown()
+			UiDropDown.destroyDropDown(self)
+			enableSaveLoad(true)
+		end
+		--- localized earlier before savePreset
+		function setPreset(id)
+			presetDropdown.value = id
+		end
+		--- loads the current preset
+		size = size / 2
+		local loadPresetButton = Ui()
+			:widthpx(WEAPON_WIDTH * size):heightpx(BUTTON_HEIGHT)
+			:settooltip(GetText("ConfigureWeaponDeck_PresetLoad_Tooltip"))
+			:decorate({
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_PresetLoad_Title")),
+			})
+			:addTo(buttonLayout)
+		function loadPresetButton.onclicked()
+			loadPreset(presetDropdown.value)
+			enableSaveLoad(false)
+			return true
+		end
+		--- Saves the current preset
+		local savePresetButton = Ui()
+			:widthpx(WEAPON_WIDTH * size):heightpx(BUTTON_HEIGHT)
+			:settooltip(GetText("ConfigureWeaponDeck_PresetSave_Tooltip"))
+			:decorate({
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_PresetSave_Title")),
+			})
+			:addTo(buttonLayout)
+		savePresetButton.disabled = true
+		function savePresetButton.onclicked()
+			local value = presetDropdown.value
+			if value ~= PRESET_DEFAULT then
+				savePreset(value)
+			end
+			enableSaveLoad(false)
+			return true
+		end
+		--- Define function to enable/disable the buttons, localized earlier
+		function enableSaveLoad(enable)
+			local value = presetDropdown.value
+			if enable then
+				-- vanilla and random cannot save, new cannot load
+				savePresetButton.disabled = value == PRESET_DEFAULT or value == PRESET_RANDOM
+				loadPresetButton.disabled = value == PRESET_NEW
+			else
+				-- random can always load
+				savePresetButton.disabled = true
+				loadPresetButton.disabled = value ~= PRESET_RANDOM
+			end
+		end
+
+		-------------
+		-- Weapons --
```

Comment:
Generally it might be a good idea to separate such sections into functions - helps with readability